pkg/kubectl/cmd/edit.go

```diff
@@ -151,24 +150,32 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 	}
 
 	encoder := f.JSONEncoder()
-
-	windowsLineEndings := cmdutil.GetFlagBool(cmd, "windows-line-endings")
-	edit := editor.NewDefaultEditor(f.EditorEnvs())
 	defaultVersion, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
 	if err != nil {
 		return err
 	}
-	results := editResults{}
-	for {
-		objs, err := resource.AsVersionedObjects(infos, defaultVersion.String(), encoder)
-		if err != nil {
-			return preservedFile(err, results.file, out)
-		}
-		// if input object is a list, traverse and edit each item one at a time
-		for _, obj := range objs {
-			// TODO: add an annotating YAML printer that can print inline comments on each field,
-			//   including descriptions or validation errors
+	objs, err := resource.AsVersionedObjects(infos, defaultVersion.String(), encoder)
+	if err != nil {
+		return err
+	}
 
+	var (
+		windowsLineEndings = cmdutil.GetFlagBool(cmd, "windows-line-endings")
+		edit               = editor.NewDefaultEditor(f.EditorEnvs())
+		results            = editResults{}
+		original           = []byte{}
+		edited             = []byte{}
+		file               string
+	)
+
+outter:
+	for i := range objs {
+		obj := objs[i]
```

Comment:
Ever since bugs like https://github.com/kubernetes/kubernetes/pull/18331 or https://github.com/kubernetes/kubernetes/pull/18321 I have stopped using this syntax when ranging over references:)